gammapy/time/exptest.py

```diff
@@ -0,0 +1,41 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import, division, print_function, unicode_literals
+import numpy as np
+
+__all__ = [
+    'exptest_for_run',
+]
+
+def exptest_for_run(time_delta):
+    """Compute Mr value, the level of variability for a certain period of time.
+
+     A single Mr value can be calculated, which shows the level of variability
+     for the whole period, or the Mr value for each run can be shown.
+
+
+    Ref:Prah(1999),http://adsabs.harvard.edu/abs/1999astro.ph..9399P
+
+    Parameters
+    ----------
+    time_delta : list of float
+        the time difference between two adjacent events
+
+    Returns
+    -------
+    Mr : float
+        Level of variability
+    """
+    mean_time = np.mean(time_delta)
+    normalized_time_delta = time_delta / mean_time
+    sum_time = []
```

Comment:
I haven't looked at your code in detail, but can't you avoid the for loop and do the same computation using just Numpy array expressions roughly like this?
```python
mask = normalized_time_delta < 1
sum_time = 1 - normalized_time_delta[mask] / 1
```
This is using "fancy indexing using boolean arrays":
http://www.scipy-lectures.org/intro/numpy/array_object.html#using-boolean-masks

There's other Numpy tricks that will usually let you avoid Python for loops and get much better performance.
Let me know if this doesn't work, then I'll have a closer look.